### PR TITLE
fix: handle the initial empty signature when signing documents

### PR DESCRIPTION
**Pull request by felixcicatt**
When first uploading a document, OpenZaak initializes it with an empty signature (not null). This is now supported by ZAC when signing documents.

Solves PZ-1584

### CODE REVIEW

File: src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java [hunk 1]
```diff
@@ -684,9 +686,14 @@ public Response ondertekenInformatieObject(
...
-                     policyService.readDocumentRechten(enkelvoudigInformatieobject,
-                             zrcClientService.readZaak(zaakUUID)).ondertekenen());
+        Zaak zaak = zrcClientService.readZaak(zaakUUID);
+        DocumentRechten documentRechten = policyService.readDocumentRechten(
+                enkelvoudigInformatieobject,
+                zaak
+        );
+        Ondertekening ondertekening = enkelvoudigInformatieobject.getOndertekening();
+        boolean hasOndertekening = ondertekening != null && ondertekening.getDatum() != null;
+        assertPolicy(!hasOndertekening && documentRechten.ondertekenen());
```
Comment (edgarvonk) [inline src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java:696]: I would suggest to move `policyService.readDocumentRechten(
                enkelvoudigInformatieobject,
                zaak
        );` back into the `assertPolicy` call. The reason is that we want to keep the policy logic in one place as much as possible. Ideally we even move some of this logic to the OPA Rego policies but that is for another day I would say. :-)
        
```suggestion
        assertPolicy(
           !hasOndertekening && policyService.readDocumentRechten(
                enkelvoudigInformatieobject,
                zaak
            ).ondertekenen()
        );
```
        
ps: you may need to reformat the code after applying this suggestion..

File: src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java [hunk 2]
```diff
@@ -684,9 +686,14 @@ public Response ondertekenInformatieObject(
     ) {
         final EnkelvoudigInformatieObject enkelvoudigInformatieobject = drcClientService.readEnkelvoudigInformatieobject(
```
Comment (edgarvonk) [inline src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java:687]: Some coverage by a unit and/or integration test would be great of course. Not sure if that is feasible in the time left for this sprint..?

Comment (felixcicatt) [reply src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java:687]: Fair point, I'll give it a go

**Review by bas-info-nl — APPROVED** [review]

File: src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java (hunk 2 above)
Comment (felixcicatt) [reply src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java:687]: I added an integration test :)

**Pull request merged**